### PR TITLE
CARGO_TARGET_DIR test

### PR DESCRIPTION
**Pull request by TriplEight**
Highly experiment.

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -13,8 +13,10 @@ stages:
 variables:
   GIT_STRATEGY:                    fetch
   CARGO_HOME:                      "/ci-cache/${CI_PROJECT_NAME}/cargo/${CI_JOB_NAME}"
+  # CARGO_TARGET_DIR:                "/ci-cache/${CI_PROJECT_NAME}/target/${CI_JOB_NAME}"
```
Comment (Robbepop) [inline .gitlab-ci.yml]: Please remove instead of commenting out. We have git history anyway to restore if needed.

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -13,8 +13,10 @@ stages:
...
+  CARGO_TARGET_DIR:                "/ci-cache/${CI_PROJECT_NAME}/targets/${CI_COMMIT_REF_NAME}/${CI_JOB_NAME}"
   SCCACHE_DIR:                     "/ci-cache/${CI_PROJECT_NAME}/sccache"
-  CARGO_INCREMENTAL:               0
+  # CARGO_INCREMENTAL:               0
```
Comment (Robbepop) [inline .gitlab-ci.yml]: same here

File: .gitlab-ci.yml [hunk 3]
```diff
@@ -61,6 +64,10 @@ check-std:
   stage:                           check
   <<:                              *docker-env
   script:
+    # debug
+    - echo $CARGO_TARGET_DIR
+    # - du -sh /ci-cache/${CI_PROJECT_NAME}/target/*
```
Comment (Robbepop) [inline .gitlab-ci.yml]: also here

File: .gitlab-ci.yml [hunk 4]
```diff
@@ -78,7 +85,9 @@ check-wasm:
 build-std:
   stage:                           workspace
   <<:                              *docker-env
-  dependencies:
+  # dependencies:
+  #   - check-std
+  needs:
     - check-std
```
Comment (Robbepop) [inline .gitlab-ci.yml]: What is the semantics of this? Can you shortly explain? And what is the difference to the former `dependencies`?

Comment (TriplEight) [reply .gitlab-ci.yml]: It launches right after the mentioned job succeeds, doesn't wait for it's stage to be over.

**Pull request merged**